src/pages/Dashboard/Event.tsx

```diff
@@ -0,0 +1,164 @@
+    const navigateTo = useNavigate();
+
+    function checkMember() {
+        try {
+            const loggedIn = localStorage.getItem('member_id') !== null;
+            const id = localStorage.getItem('member_id') ?? undefined;
+            const displayName = localStorage.getItem('member_name') ?? "guest";
+
+            setIsLoggedIn(loggedIn);
+            setMemberId(id);
+            setDisplayName(displayName);
+        } catch (error) {
+            console.error('Error checking user status (member/nonmember): ', error);
+        }
+    }
+
+    async function fetchEvent() {
+        try {
+            const response = await fetch(`${import.meta.env.VITE_API_URL}/api/v1/events/${event_id}`, {
+                method: 'GET',
+                headers: {
+                    'Content-Type': 'application/json',
+                }
+            });
+
+            if (!response.ok) {
+                throw new Error('Network response was not ok');
+            }
+            const data: eventType = await response.json();
+            setEvent({
+                ...data,
+                date: new Date(data.date),
+            });
+            console.log(data);
+        } catch (error) {
+            console.error('Error fetching event:', error);
+        } finally {
+            setLoading(false);
+        }
+    }
+
+    useEffect(() => {
+        checkMember();
+        fetchEvent();
+    }, [event_id]);
+
+    if (loading) return <p>Loading...</p>;
+    if (!event) return <p>No event details available.</p>;
+
+    return (
+        <div className="background-event">
+            <div className="header">
+                <div className="header-icon">
+                    <a href="/"><img src="./src/assets/pmc_logo.svg" className="logo" ></img></a>
+                </div>
+                <nav className="header-nav">
+                    <a href="#upcoming-events" className="header-link">Events</a>
+                    <div>
+                        {isLoggedIn ? (
+                            <a href="/profile" className="header-link">Profile</a>
+                        ) : (
+                            <a href="/" className="header-link">Profile</a>
+                        )}
+                    </div>
+                    {/* <a href="/" className="header-link">Profile</a> */}
+                </nav>
+            </div>
+
+
+
+            <div className="event-container">
+                <h2 className="event-title">{event.name}</h2>
+                <div className="event-details-container">
+                    <div className="event-details">
+                        <p>{event.date.toDateString()}</p>
+                        <p>{event.location}</p>
+                        <p>member price: ${event.member_price !== undefined ? event.member_price.toFixed(2) : "N/A"}</p>
+                        <p>non-member price: ${event.non_member_price !== undefined ? event.non_member_price.toFixed(2) : "N/A"}</p>
```

Comment:
It looks like member price always covers the non-member price.

Reply:
If the event is member_only, we can display just the member price. Otherwise, we will display both.